iso15118/shared/exi_codec.py

```diff
@@ -383,6 +387,18 @@ def from_exi(
                 f"Error parsing the decoded EXI into a Pydantic class: {exc}. "
                 f"\n\nDecoded dict: {decoded_dict}"
             ) from exc
+        except V2GMessageValidationError as exc:
+            if namespace == Namespace.ISO_V2_MSG_DEF:
+                exc.message = V2GMessageV2.construct_skip_validation(
+                    **decoded_dict["V2G_Message"]
+                )
+            elif namespace == Namespace.DIN_MSG_DEF:
+                exc.message = V2GMessageDINSPEC.construct_skip_validation(
+                    **decoded_dict["V2G_Message"]
+                )
+            elif namespace.startswith(Namespace.ISO_V20_BASE):
+                exc.message = msg_class.construct(**msg_dict)
+            raise exc
```

Comment:
this I guess is solving part of the problem but not all of it.
The Pydantic `parse_obj` method will try to construct a python class based on the dictionary message we pass to it...

Now, if that parsing fails, a `ValidationError` is raised (https://pydantic-docs.helpmanual.io/usage/validators/) and currently it seems the way we deal with it is this:
```python
raise EXIDecodingError(
                f"Error parsing the decoded EXI into a Pydantic class: {exc}. "
                f"\n\nDecoded dict: {decoded_dict}"
            ) from exc
```

so maybe here we need to raise a `V2GMessageValidationError` instead?

Reply:
Done.